src/algorithms/structures/extended_node.py

```diff
@@ -0,0 +1,65 @@
+from collections import OrderedDict
+from itertools import chain
+
+import numpy as np
+import bblfsh
+import json
+
+
+class ExtNode(object):
+    """
+    Extended Node used for efficiently extract paths from a tree structure
+
+    bn, depth, numeration: are initialized during tree top-down traversal
+    children: during the recursive return call
+    log_parents: annotated in a separate traversal
+
+    Attributes:
+        bn (*): original tree node
+        depth (int): depth of the node in the tree
+        numeration (int): if node is a leave indicates the position from left to right (0-indexed), otherwise it's -1
+        children (list(ExtNode)): list of the extended children of 'bn'
+        log_parents (list(ExtNode)): list of the log_parents used to compute the least common ancestor (lca)
+
```

Comment:
you can put comments about attributes to __init__ doc string and types information to:
```python
def __init__(self, base_node: ExtNode, depth: int, numeration: int=-1):
```